documentation/docs/tutorials/single-topic-multi-tenant.mdx

```diff
@@ -0,0 +1,341 @@
+```
+
+OPAL Client logs:
+```bash
+docker compose -f docker-compose-single-topic-multi-tenant.yml logs -f opal_client
+```
+
+Look for these log patterns:
+
+**Server side:**
+```
+Publishing data update to topics: {'tenant_data'}
+Broadcasting incoming event: {'topic': 'tenant_data'}
+```
+
+**Client side:**
+```
+Received notification of event: tenant_data
+Fetching data from url: http://example_external_data_provider:80/acl/tenant1
+Saving fetched data to policy-store: destination path='/acl/tenant1'
+```
+
+## Data Updates
+
+### Update existing tenant data
+
+```bash
+curl -X POST http://localhost:7002/data/config \
+  -H "Content-Type: application/json" \
+  -d '{
+    "entries": [{
+      "url": "http://example_external_data_provider:80/acl/tenant1",
+      "topics": ["tenant_data"],
+      "dst_path": "/acl/tenant1"
+    }],
+    "reason": "Refresh tenant1 data"
+  }'
+```
+
+## Testing Authorization
+
+### Test tenant-specific policies
+
+```bash
+curl -X POST http://localhost:8181/v1/data/policies/rbac/allow \
+  -H "Content-Type: application/json" \
+  -d '{
+    "input": {
+      "user": "alice",
+      "action": "read",
+      "resource": "document1",
+      "tenant_id": "tenant1"
+    }
+  }' | jq .
+```
+
+## Production Considerations
+
+### Use webhooks instead of polling
+
+For production environments, configure webhooks for immediate policy updates:
+
+```yaml
+opal_server:
+  environment:
+    - OPAL_POLICY_REPO_WEBHOOK_SECRET=your-secure-secret
+```
+
+### Secure data sources
+
+Configure authentication for external data APIs:
+
+```bash
+curl -X POST http://localhost:7002/data/config \
+  -H "Content-Type: application/json" \
+  -d '{
+    "entries": [{
+      "url": "https://secure-api.yourdomain.com/tenant-data/tenant1",
+      "config": {
+        "headers": {
+          "Authorization": "Bearer your-secure-token"
+        }
+      },
+      "topics": ["tenant_data"],
+      "dst_path": "/acl/tenant1"
+    }]
+  }'
+```
+
+### Scale horizontally
+
+Increase worker count for higher load:
+
+```yaml
+opal_server:
+  environment:
+    - UVICORN_NUM_WORKERS=4
+```
+
+## Troubleshooting
+
+### Tenant not receiving data
+
+Check:
+1. Topic configuration: `OPAL_DATA_TOPICS=tenant_data`
+2. External data provider endpoint availability
+3. OPAL Client logs for connection errors
+
+```bash
+# Test data endpoint
+curl http://localhost:8090/acl/tenant1
+
+# Check client logs
+docker logs $(docker compose ps -q opal_client)
+```
+
+### Data isolation issues
+
+Verify:
+1. Unique `dst_path` for each tenant
+2. Policies correctly handle `tenant_id` in input
+
+```bash
+# Check OPA data structure
+curl -s http://localhost:8181/v1/data | jq .
+```
+
+## Comparison with Traditional Approach
+
+| Aspect | Traditional | Single-Topic |
+|--------|-------------|--------------|
+| Topics per 1000 tenants | 1000 | 1 |
+| Service restart required | Yes | No |
+| Memory usage growth | O(N²) | O(N) |
+| Configuration complexity | High | Low |
```

Comment:
This is a duplicate of the info stated above, and contain some vague terms. I would omit that from the tutorial as it is not needed.

```suggestion
```